### DISCUSSION TITLE
Databases: Add Snowflake support

### DISCUSSION THREAD

**Discussion by mstui** [directus/directus#14173] [Feature Requests]
Hi all,

we plan to use Directus for reference data management within our analytics tool stack. Our data warehouse is based on [Snowflake](https://www.snowflake.com), which is (among other features) a standard SQL compatible database. Since Directus claims to support "any SQL database", I'd like to request Directus support for Snowflake.

In our use case, this would help us to avoid administration effor and costs for a second database instance (Postgres in our case) which is currently required only for directus. Additionally, it would be much easier for us to get data in and out to/from directus with our standard data warehouse tools and processes.

As I understand, Directus leverages "Knex" for query building and database connections. There seems to be a [Snowflake "dialect" for Knex](https://github.com/deliverr/data-knex-snowflake-dialect#readme). Maybe that could be a starting point for further investigation on what is needed to add Snowflake support to Directus.

Please let me know what I can help to support this request further. Thanks for looking into this.

Michael

**Comment by mstui** [comment]
Snowflake recently introduced their [Unistore ](https://www.snowflake.com/blog/introducing-unistore/) concept. This makes Snowflake more usable for transactional data and would support a native Directus connector even better.

**Reply by rijkvanzanten** [reply]
It's a little difficult to support these more vendor specific features. For example, the SQL query `CREATE HYBRID TABLE x ()` will fail in every database that's not Snowflake 🤔

**Comment by bastsack** [comment]
Any news regarding a direct integration to snowflake databases? Would be awesome

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
